homeassistant/components/tuya/sensor.py

```diff
@@ -138,6 +138,21 @@
         ),
         *BATTERY_SENSORS,
     ),
+    # Irrigator
+    # https://developer.tuya.com/en/docs/iot/categoryggq?id=Kaiuz1qib7z0k
+    "ggq": (
+        SensorEntityDescription(
+            key=DPCode.COUNTDOWN,
+            name="Countdown",
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key=DPCode.COUNTDOWN_LEFT,
+            name="Countdown left",
+            state_class=STATE_CLASS_MEASUREMENT,
```

Comment:
This counts down the number of seconds. We don't allow that in sensors, instead calculate the datetime of when the timer finishes and use that (give it the `timestamp` device class as well).

Additionally, this isn't a measurement?

Reply:
Actually it counts down the minutes left.

countdown_left | Remaining time of countdown | Integer | {“unit”:“min”,“min”:0,“max”:1440,“scale”:0,“step”:1}

I am not sure how to add this one correctly.

Reply:
@frenck can you give me a hint or support on calculation of the datetime when the timer finishes? Currently I am using it as a sensor and it updates every minute until 0. Indeed it is not bad to see how much minutes are left instead of a timestamp.

Reply:
Calculate a datetime instead and return that :)